pkg/operator/configobserver/oauth/observe_tokenconfig_test.go

```diff
@@ -0,0 +1,160 @@
+package oauth
+
+import (
+	"fmt"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/api/equality"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/util/diff"
+	"k8s.io/client-go/tools/cache"
+
+	configv1 "github.com/openshift/api/config/v1"
+	configlistersv1 "github.com/openshift/client-go/config/listers/config/v1"
+	"github.com/openshift/library-go/pkg/operator/events"
+	"github.com/openshift/library-go/pkg/operator/resourcesynccontroller"
+)
+
+type testLister struct {
+	lister configlistersv1.OAuthLister
+}
+
+func (l testLister) OAuthLister() configlistersv1.OAuthLister {
+	return l.lister
+}
+
+func (l testLister) ResourceSyncer() resourcesynccontroller.ResourceSyncer {
+	return nil
+}
+
+func (l testLister) PreRunHasSynced() []cache.InformerSynced {
+	return nil
+}
+
+func TestObserveTokenConfig(t *testing.T) {
```

Comment:
(No action required) I don't see all paths being tested. Maybe test corner cases of the helper functions (`ObserveAccessTokenInactivityTimeoutSeconds` and `ObserveAccessTokenMaxAgeSeconds`) independently to make this easier?

Reply:
I have the same thought. I will try to add separate tests for paths that are not tested.

Reply:
I refactored the code and added few tests. Now the coverage is close to 100%.